### ISSUE TITLE
Runtime control for batch settings

### ISSUE THREAD

**Issue by michaelschaarschmidt** [tensorflow/serving#292]
Hello,

I am trying to understand the feasibility of implementing a more sophisticated runtime controller for inference scheduling. To start off, I am trying to work out what exactly this would entail in terms of what the serving API allows/what I would need to modify.

A contrived example setting:

- There is a heteregenous (but for now fixed) infrastructure available with a number of servers, some possessing GPUs, some only CPUs.
- Multiple TensorFlow servables, each occasionally changing versions
- Non-stationary access distribution of inference requests of different models/their versions
- Trying to optimise overall inference throughput and resource utilisation

The way I understand this is that one would have to predefine batching parameters per model according to the tuning guide and then use the _SharedBatchScheduler_ (or the _StreamingBatchScheduler_ for mixed workloads), basically hand-constructing the mixture of CPU/GPU operation assignments per model.

What I don't understand yet is how exactly the routing process to distributed devices would work in this scenario, i.e. how does serving integrate with distributed tensorflow? Using Kubernetes as described in the tutorial, it seems like every serving instance would just interact with the device in its container. So to be concrete:

1. For the above scenario, what would the 'typical' deployment look like? 
2. Is it correct that serving is meant to be used with separate serving instances per cluster node, as in, no centralised control? 

Basically, I am trying to assess at what level runtime control could be injected (i.e., how many devices a single serving instance is supposed to control), what parameters can realistically be adjusted, and which have to be compiled statically due to the way models are loaded and executed by the underlying tf graph scheduler.

**Comment by harshini-gadige** [comment]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow-serving

If you open a GitHub issue, it must be a bug, a feature request, or a
significant problem with documentation (for small docs fixes please send
a PR instead).

Thanks!
